---
title: "Balanus example - Molecular Ecology"
author: "John Wares"
date: "1/25/2021"
output: html_document
---

# Why did I put this together?

When you learn how to do this work in a coding environment like **R**, it isn't just about doing it the nerdy way. Coding scientific research makes it more *repeatable* and *reproducible*. I can go back to some of the analyses from my paper that I finished working on 2 years ago and repeat the analyses exactly -- or, if the data changed (say, I added more data), then all of the analyses could be done on the updated data *and the figures would be re-made using the same specifications as before*. 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(adegenet)
library(poppr)
library(pegas)
library(hierfstat)
library(diveRsity)
```

## Your assignment

Assuming I put all this together correctly, you should have this .Rmd file and the file May19.snps.gen, accessed via code below. Please note: I'm not a highly skilled coder, there are definitely more sophisticated approaches to all of the outcomes below. However, using this document you can run through each of the analytical steps individually by pressing the 'play' button at the top right of each 'code chunk'.

```{r databasic, include=FALSE, echo=FALSE}

file3 = "May19.snps.gen"
small_Bg <- read.genepop(file3,ncode=2) 
# now no minor alleles less than 0.05 frequency. removed in "Populations" in STACKS


locus50 <- missingno(small_Bg, type="loci", cutoff=0.5) #removes 14 loci

Bg50 <-missingno(locus50, type="geno", cutoff=0.5) #1500 loci, 142 indivs, 10pops

```

In the data transformation above, the decision has been made to rather liberally include loci because there was a large amount of missing data across this sequencing run. So, loci are included with up to 50% missing information from individuals, and individuals included with up to 50% missing information from all loci. These decisions were initially made after considerable data exploration to retain as many individuals and loci as seemed useful for the question; different questions may require different analytical cutoffs. The data have been sampled from the locations on this map:

```{r maps, fig.height=7, fig.width=4.1, echo=FALSE, fig.align='center', fig.cap="Figure 1. Map of sampled locations."}
#mapping for Bglandula paper
library("maps")
library("mapproj")
library("mapdata")
data(worldMapEnv)

# all location data here, have re-checked Linda Mar, re-ordered for later analyses
locs<-c("JUN","FHL","MEA","PSG","MEN","BOD","LIN","TPT","HOP","SHB","GOL")
lats<-c(58.3,48.5,45.48,41.7,40.5,38.33,37.61,36.94,36.62,35.16,34.4)
longs<-c(-134.4,-123.01,-123.96,-124.25,-124.41,-123.04,-122.48,-122.06,-121.904,-120.69,-119.84)

locpoints<-cbind(longs,lats)

# for plotting labels
rightlats<-lats[-c(1,8)]
rightlongs<-longs[-c(1,8)]
rightlocs<-locs[-c(1,8)]
leftlats<-lats[c(1,8)]
leftlongs<-longs[c(1,8)]
leftlocs<-locs[c(1,8)]
PSGlats<-lats[c(4)]
PSGlongs<-longs[c(4)]
PSG<-cbind(PSGlongs,PSGlats)



#pdf("MapFig1.pdf")

m0 <- maps::map(database = "world", plot = TRUE, fill=FALSE, xlim = c(-140,-118), ylim = c(30,60),lforce="n")


points(locpoints,cex=2,pch=24,col="darkblue",bg="steelblue",xlab="longitude",ylab="latitude")
#points(PSG,cex=2,pch=17,col="black")
#points(PSG,cex=3,pch=4,col="black")
text(rightlongs,rightlats,rightlocs,pos=2,offset=1.5,cex=1)
text(leftlongs,leftlats,leftlocs,pos=4,offset=1,cex=1)


#dev.off()

```

However, note that one site (PSG) no longer has any individuals included; the libraries from that source were mishandled in the lab by JPW, apparently, and contributed most of the 'missing' information (all sorts of surprises in this kind of work, pay attention in the lab! In this case it is likely that I did not correctly dilute the DNA concentration for those samples) **The organism is an acorn barnacle :) meaning adults are sessile, offspring can drift for 10s of kilometers before they settle.** What questions does this raise about how the ocean environment will influence the distribution of diversity in this species?

Let's first take a close look at the data. Here we have SNPs generated using a restriction-site limited sequencing effort, meaning that only a small subset of the overall genome is sequenced - and thus enough sequences cover each part of each genome that we can determine not only the sequence of each fragment but whether there are polymorphic sites within a fragment. But think about that carefully from the point of view of how you *know* each fragment that is sequenced, in one individual, is the same or different from a similar fragment sequenced in another individual?

In Chapter 4, we started thinking about F-statistics, or the broad family of hierarchical statistics that use different models depending on the marker type to compare diversity in nested 'containers' of diversity. I need to go back and unpack one of them a bit. The *inbreeding coefficient* we explored can also be evaluated as Xis: the heterozygosity in an individual relative to the heterozygosity at the sample/subpopulation it is within. If an individual contains a lot of homozygous loci despite there being more heterozygosity in the sample, Xis is elevated - it suggests some form of inbreeding. If a locus is consistently elevated in homozygosity across all individuals in the sample, however (relative to all other loci), it may be that what we are 'seeing' is an artefact. This is a step where we have to think about the *natural history* of the markers, the process of obtaining marker data, and the *natural history* of the organism as well to consider what we might expect to see. 


```{r fstats, echo=FALSE}

# here we play with Fis and Fst as needed to figure out ways to fix the data just a bit earlier

data50<-genind2hierfstat(Bg50)
#library(hierfstat)
basic50<-basic.stats(data50)
hist(basic50$Fis)
```
So starting with our initial sample of 1500 bi-allelic SNP loci, in an organism that is believed to be an obligate outcrossing hermaphrodite, we might be very surprised to see some loci with extraordinarily high (and low) Xis values. Now refer back to Figure 2-2 - this is what we are grappling with, in a sense. When we don't have an excellent genome to assemble sequenced fragments onto, some of the decisions that are made in the bioinformatic steps that generate these loci from raw sequence data can cause two problems: (1) a locus that is real but highly diverse, e.g. many SNPs in the region that was sequenced, may be split into multiple loci - each of which is less diverse, or (2) loci that are *actually distinct*, in different locations in the genome, but which are similar in sequence identity perhaps because of gene region duplications, may be combined and diversity is noted that is not actually inherited in a Mendelian way. Of course, a broad range of variation in a metric like Fis is expected because of the process of sampling individuals and loci, some of which *may* be influenced by other evolutionary mechanisms!

So, choose your own adventure: what do you do now?

1. go back and attempt to rebuild the SNP dataset with different parameters and hope that fixes some of the problems
2. evaluate how to exclude loci that are still behaving in surprising ways, given that you cannot know the truth about any single locus without a tremendous amount of extra work and/or having a well-constructed genome for your species.

Most importantly, take some time to think about how it is important to ***look at your data*** and learn to intuit how the data are behaving without just blindly plunging into additional analyses. Many of the iterations that are necessary in this kind of science could be avoided if you learn to think carefully about the data first!

This isn't just a problem that arises with inference of variable loci from a protocol like RAD-seq by the way. Designing primers for microsatellites or other variable loci can often lead to markers that do not appear to fit our expectations of Hardy-Weinberg equilibria, and are typically then excluded.

In the case of this data set, the resultant analyses are not highly dependent on whether the 'misbehaving' loci are included or not, but often that will not be the case so it is always good to take a deep breath and spend some time on the data before getting attached to the outcomes of any further analysis.

So, lets now look at the data having trimmed out many of the misbehaving loci:

```{r fstatstrimmed, echo=FALSE}

Fiscut25<-read.table("FisWhitelist25.txt",as.is = TRUE) 

Fiscut25<-as.vector(Fiscut25)
Bg50cut25<-Bg50[,loc = Fiscut25$V1 ]
Bg50<-Bg50cut25

newdata50<-genind2hierfstat(Bg50)
#library(hierfstat)
newbasic50<-basic.stats(newdata50)
hist(newbasic50$Fis)
```

Basically here I have made some decisions about trimming out some loci based on elevated inbreeding coefficients, but not all of them - because at this point, the markers are being looked at across all spatial samples from Alaska to Southern California! If we want to think about patterns of divergence among sites, we would *expect* some loci to be entirely homozygous in some individuals because of where they are, even though the marker is variable across the entire data set. If you aren't sure if you understand this, go back to Chapter 4 and the discussion of $X_{st}$ - which is itself a measure of inbreeding of the sample relative to the total study!

So, decisions of how to use data are again made in the context of the questions being asked. I (JPW) tend to assume that since I cannot know everything about each locus, and having many loci will mean that any individual locus contributes very little to the pattern, I would rather proceed with as many as possible as long as they make sense. It is, admittedly, an iterative process and again - requires thinking about your data carefully. In any case, this particular response variable of our data is now more normally distributed and leaves us with 876 SNP loci.

<style>
div.rose { background-color:#ffc4ef; border-radius: 10px; padding: 40px;}
</style>
<div class = "rose">

**Questions for you to answer before next class period**

1. Write a paragraph that explains your understanding of how these data were obtained.
2. Write a paragraph that explains your understanding of how decisions in the assembly of these data can influence the outcome of our analyses.
3. Write a paragraph that explains your understanding of what patterns of data lead to low *X*st, and what lead to high *X*st - and for the data we are using here, what would we replace '*X*' with, and why?
4. Given what you know of the natural history of the organism, what questions does this raise about how the ocean environment will influence the distribution of diversity in this species? What might you predict, overall - use your own words don't worry about technical language.

This concludes your first day snooping through this data set.

</div>

Next, we are exploring overall patterns in these data. Analysis of population genomic data is an exploratory, iterative process - you often have to analyze the data in a first pass to consider what the most useful next considerations are. 

```{r pairwise, echo=FALSE}

pairwise50<-pairwise.neifst(newdata50) 
pairwise50

```

Following generation of pairwise Fst values, lets explore the data using DAPC. 

```{r dapc1, echo=FALSE}

#from Balanus, need to work it to this file and these data and k.i.s.s.

mypops<-popsub(Bg50)
#myclusters<-find.clusters(mypops,n.pca=50,n.clust=3)

#instead of the commented command above, just type in the console (after executing the mypops line):

#find.clusters(mypops)

#and explore what happens with different values for n.pca!
#what happens if you include less "eigeninformation" (fewer pca)?
#what happens if you include more?
#what do you see as the best answer for the number of clusters in these data?

dapcv1<-dapc(Bg50,var.contrib=TRUE,scale=FALSE,n.pca=50,n.da=2,truenames=FALSE)
scat_dapcv1<-scatter(dapcv1,cell=0,pch=18:23,cstar=0,mstree=FALSE,lwd=2,lty=2) #okwhewanswerissimilar
scat_dapcv1_1d<-scatter(dapcv1,1,1,scree.da=FALSE, legend=TRUE, solid=.4,bg="white")




```


Now lets try snmf.... what does that mean? "sparse non-negative matrix factorization", see http://membres-timc.imag.fr/Olivier.Francois/snmf/index.htm for more details on this. Effectively this is a computationally efficient way to estimate the "ancestry coefficients" of individuals among a set of *K* populations, as with Structure. 

WAHOOO MAKE SURE THIS CALL AND RESPONSE DOESN'T GO INTO THE HTML OUTPUT, 100s of COMMAND REPEATS... :) not sure which command, worry about that tomorrow.

This code chunk will take a few minutes - in part because it is repeating one analysis many times to help us "objectively" determine what the statistically-optimal number of populations (K) would be for these data. *So we will take this time to talk about repeatability and reproducibility in science, as well as the methods and hypotheses in this study.*

## snmf output

below will be 4 plots, and you can look at the .Rmd to find the code that got us here. The first looks like a funny dot plot and is the mechanism by which SNMF approaches help guide the scientist towards the most appropriate value of *K* ancestral populations for the data to have come from; typically the lowest point is the best, but sometimes as in this plot there may be 2 that are quite similar in their "cross-entropy".

Below that are the bar graphs for population composition of each individual, starting in the north to the left and to the south as you move to the right. First is K=2, then K=3, then K=5.



```{r snmf, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}

# just hit play on this chunk and walk away for 5-10 minutes....


#1 genind2structure.R by Lindsay Clark https://github.com/lvclark/R_genetics_conv/blob/master/genind2structure.R

# Function to export to STRUCTURE format from genind object.
# genind objects are created in the R package adegenet.  The function below is an R function.
# Lindsay V. Clark, 26 July 2015

# obj: genind object
# file: file name to write
# pops: whether to include population info in the file
# Function is flexible with regards to ploidy, although genotypes are
# considered to be unambiguous.
# Missing data must be recorded as NA in obj@tab.

# example use: 
# data(nancycats)
# genind2structure(nancycats, file="nancy_structure.txt", pops=TRUE)

genind2structure <- function(obj, file="", pops=FALSE){
  if(!"genind" %in% class(obj)){
    warning("Function was designed for genind objects.")
  }
  
  # get the max ploidy of the dataset
  pl <- max(obj@ploidy)
  # get the number of individuals
  S <- adegenet::nInd(obj)
  # column of individual names to write; set up data.frame
  tab <- data.frame(ind=rep(indNames(obj), each=pl))
  # column of pop ids to write
  if(pops){
    popnums <- 1:adegenet::nPop(obj)
    names(popnums) <- as.character(unique(adegenet::pop(obj)))
    popcol <- rep(popnums[as.character(adegenet::pop(obj))], each=pl)
    tab <- cbind(tab, data.frame(pop=popcol))
  }
  loci <- adegenet::locNames(obj) 
  # add columns for genotypes
  tab <- cbind(tab, matrix(-9, nrow=dim(tab)[1], ncol=adegenet::nLoc(obj),
                           dimnames=list(NULL,loci)))
  
  # begin going through loci
  for(L in loci){
    thesegen <- obj@tab[,grep(paste("^", L, "\\.", sep=""), 
                              dimnames(obj@tab)[[2]]), 
                        drop = FALSE] # genotypes by locus
    al <- 1:dim(thesegen)[2] # numbered alleles
    for(s in 1:S){
      if(all(!is.na(thesegen[s,]))){
        tabrows <- (1:dim(tab)[1])[tab[[1]] == indNames(obj)[s]] # index of rows in output to write to
        tabrows <- tabrows[1:sum(thesegen[s,])] # subset if this is lower ploidy than max ploidy
        tab[tabrows,L] <- rep(al, times = thesegen[s,])
      }
    }
  }
  
  # export table
  write.table(tab, file=file, sep="\t", quote=FALSE, row.names=FALSE)
}

#2. take the current genind object and convert



genind2structure(Bg50,file="Bg50v2.stru",pops=TRUE) #it is written to the directory
#Bgstruc<-read.structure("Bg50.stru")


# this is the R approximation to Structure and seems to get a lot of support from the "landscape ecology" crowd.

#install.packages(c("fields","RColorBrewer","mapplots")) 
#BiocManager::install("LEA")
#library("LEA")

source("http://membres-timc.imag.fr/Olivier.Francois/Conversion.R")

source("http://membres-timc.imag.fr/Olivier.Francois/POPSutilities.R")


#install.packages("~/Downloads/LEA_3.0.0-2.tar", repos = NULL, type ="source")

library("LEA")
library("fields","RColorBrewer","mapplots")


struct2geno(file="~/Dropbox/BgRadness/Bg50v2.stru", FORMAT=2, extra.col=1,extra.row=1,output="balstruc.geno")

#note a considerable amount of code here that I am not going to dig into with you, but sometimes you have to rely on the R/popgen community at large to provide solutions that get your data into the right format for what you need to do!

# this next little thing only takeas a few minutes but tells you the best K-means solutions by minimizing the value

obj.balK = snmf("~/Dropbox/BgRadness/balstruc.geno", K = 1:10, ploidy = 2, entropy = T, repetitions=20,project = "new")
plot(obj.balK, col = "blue4", cex = 1.4, pch = 19) #so... how many "populations" exist here according to this approach?

obj.bal2<-snmf("~/Dropbox/BgRadness/balstruc.geno", K=2,alpha=50,project="new")
qmatrix.bal2<-Q(obj.bal2)

barplot(t(qmatrix.bal2),col=c("darkblue","lightblue"))

obj.bal3<-snmf("~/Dropbox/BgRadness/balstruc.geno", K=3,alpha=50,project="new")
qmatrix.bal3<-Q(obj.bal3)

barplot(t(qmatrix.bal3),col=c("darkblue","lightblue","green"))

# use this last example, replacing '5' with arbitrary values from 2 to 10, to convince yourself what population structure seems USEFUL to describe to the world!

obj.bal5<-snmf("~/Dropbox/BgRadness/balstruc.geno", K=5,alpha=50,project="new")
qmatrix.bal5<-Q(obj.bal5)

barplot(t(qmatrix.bal5),col=c("darkblue","lightblue","green","yellow","pink")) #nb I chose these colors to be a bit garish bc I think K=5 is not great but if you have K>5 you need a larger list of colors, that means you need to Google to understand color terms in R :)

```


So, that isn't the whole set of analyses that were done. If you want to know more, the paper is at http://sotkae.people.cofc.edu/Publications_files/WaresStrandSotka2021.pdf where you can directly get the PDF. The focus here was on finding not just that there *is* population spatial structure of genomic data, but identifying some of the mechanisms - which appear to be related to a region of low habitat availability interacting with broader environmental gradients. Are they two distinct species? No - but they may still function distinctly in some way in these ecosystems. 

***What would you like to add to this base analysis to address questions you may have with multilocus SNP data? Would there be a spatial subset that merits more exploration?***